snapshots/overlay/overlayutils/check.go

```diff
@@ -167,3 +169,100 @@ func NeedsUserXAttr(d string) (bool, error) {
 	}
 	return true, nil
 }
+
+const (
+	MetacopyDefault     = 0
+	MetacopySupported   = 1
+	MetacopyUnsupported = 2
+	MetacopyCheckFailed = 3
+)
+
+func SupportsMetacopy(root string) (int, error) {
+	metacopyState := MetacopyCheckFailed
+
+	if _, err := os.Stat("/sys/module/overlay/parameters/metacopy"); err == nil {
+		metacopy, err := os.ReadFile("/sys/module/overlay/parameters/metacopy")
```

Comment:
It seems the stat is not used. Can't we just remove the first if to stat, and try to read the file? If it doesn't exist, then it should return an error

Reply:
If we don't want to support the case when metacopy enabled by default in Linux kernel (CONFIG_OVERLAY_FS_METACOPY=Y) and we would like to disable it by passing metacopy=off to overlay mount point we can do like this.

Reply:
I think this will become more clear once you push the next iteration. We can discuss it then, if needed :)